src/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/Context.cs

```diff
@@ -1242,11 +1242,14 @@ private static ContextOptions GetDefaultOptionForStore(ContextType storeType)
             {
                 return DefaultContextOptions.MachineDefaultContextOption;
             }
-            else
+            else if (storeType == ContextType.Domain || storeType == ContextType.ApplicationDirectory)
             {
-                Debug.Assert(storeType == ContextType.Domain || storeType == ContextType.ApplicationDirectory);
                 return DefaultContextOptions.ADDefaultContextOption;
             }
+
+            // The actual value of this is unimportant, as an invalid contextType will cause an
+            // exception to be thrown in the constructor later.
+            return ContextOptions.Negotiate;
```

Comment:
Fair enough. I reverted this change, and instead just removed the faulty debug assert